core/frontend/src/store/notifications.ts

```diff
@@ -0,0 +1,74 @@
+import { CumulatedNotification, Notification, NotificationLevel } from '@/types/notifications'
+import { Module, Mutation, VuexModule } from 'vuex-module-decorators'
+import store from '@/store'
+
+@Module({
+  dynamic: true,
+  store,
+  name: 'notification_store',
+})
+
+@Module
+export default class NotificationStore extends VuexModule {
+  notifications: Notification[] = []
+
+  cumulated_notifications: CumulatedNotification[] = []
+
+  dismissed_notifications_hashes: string[] = []
```

Comment:
while I don't think it will matter in this application, this should probably be a set instead. it is more efficient in checking if an element is present or not.

Reply:
Didn't know javascript had sets. Will change.

Reply:
For some reason I don't understand yet Vuex is not detecting changes to Set instances, thus not updating the DOM automatically.

I will keep using arrays for now and try to do some research on that.